pandas/tseries/period.py

```diff
@@ -271,10 +271,13 @@ def _from_arraylike(cls, data, freq, tz):
 
     @classmethod
     def _simple_new(cls, values, name=None, freq=None, **kwargs):
-        if not getattr(values, 'dtype', None):
+
+        if not com.is_integer_dtype(values):
```

Comment:
hmm:

```
if com.is_integer_dtype(values):
     values = np.array(values, dtype='int64', copy=False)
else:
    .... what you have below
```

I *think* that we keep the data (internally) as int64, not real sure if that is a guarantee or not.

cc @sinhrks